bin/ocaml/utop.ml

```diff
@@ -21,25 +27,47 @@ let term =
   let dir = Common.prefix_target common dir in
   if not (Path.is_directory (Path.of_string dir))
   then User_error.raise [ Pp.textf "cannot find directory: %s" (String.maybe_quoted dir) ];
-  let env, lib_config, utop_path, requires =
+  let ( env
+      , lib_config
+      , utop_path
+      , requires
+      , utop_findlib_conf
+      , utop_dev_tool_lock_dir_exists )
+    =
     Scheduler.go_with_rpc_server ~common ~config (fun () ->
       let open Fiber.O in
       let* setup = Import.Main.setup () in
       build_exn (fun () ->
         let open Memo.O in
         let* setup = setup in
+        let* () = lock_utop_if_dev_tool_enabled () in
+        let* utop_dev_tool_lock_dir_exists =
+          Memo.Lazy.force Utop.utop_dev_tool_lock_dir_exists
+        in
```

Comment:
Would it make sense to put these down nearer to where they are used? (E.g. at line 64?)